server/config/index.js

```diff
@@ -7,18 +9,46 @@ const data = require( './parser' )( configPath, {
 } );
 
 /**
- * Return config `key`.
- * Throws an error if the requested `key` is not set in the config file.
+ * Returns configuration value for given key
+ *
+ * If the requested key isn't defined in the configuration
+ * data then this will report the failure with either an
+ * error or a console warning.
+ *
+ * When in the 'development' NODE_ENV it will raise an error
+ * to crash execution early. However, because many modules
+ * call this function in the module-global scope a failure
+ * here can not only crash that module but also entire
+ * subsystems within Calypso. Therefore if the NODE_ENV is
```

Comment:
What if we simplify it a bit?

```However, because many modules call this function in the module-global scope early in execution, this will likely break main application flows or cause unexpected behavior to occur.```